tests/unit/test_matrix.py

```diff
@@ -244,19 +245,83 @@ def it_knows_how_to_assemble_its_blocks(self, request):
             [bs._subtotal_rows, bs._intersections],
         ]
 
+    @pytest.mark.parametrize(
+        ("nrows", "column_subtotals", "expected"),
+        (
+            (3, [], np.empty((3, 0), dtype=np.float64)),
+            (3, [1, 2], np.array([[1, 1], [2, 2], [3, 3]])),
+        ),
+    )
+    def it_knows_its_subtotal_columns(
+        self,
+        _column_subtotals_prop_,
+        _nrows_prop_,
+        _subtotal_column_,
+        nrows,
+        column_subtotals,
+        expected,
+    ):
+        _column_subtotals_prop_.return_value = column_subtotals
+        _nrows_prop_.return_value = nrows
+        bs = _BaseSubtotals(None, None)
+        assert bs._subtotal_columns.tolist() == expected.tolist()
+
+    # fixture components ---------------------------------------------
+
+    @pytest.fixture
+    def _subtotal_column_(self, request):
+        return method_mock(
+            request,
+            _BaseSubtotals,
+            "_subtotal_column",
+            return_value=np.array([1, 2, 3]),
```

Comment:
It's generally inadvisable to place return values on fixture components. That limits their reuse (and they shouldn't exist unless they're being used more than once, inlined in the test otherwise). Also it separates the return value from the test requiring you to scroll up and down to see what should be coming back. Generally just make a simple mock here and assign a return value in each test that uses it.

Reply:
this also got fixed. notion taken